packages/rspack/src/Compiler.ts

```diff
@@ -843,68 +843,80 @@ class Compiler {
 		this.#moduleExecutionResultsMap.set(id, executeResult);
 	}
 
+	#decorateUpdateDisabledHooks(jsTaps: binding.JsTap[]) {
+		if (jsTaps.length > 0) {
+			const last = jsTaps[jsTaps.length - 1];
+			const old = last.function;
+			last.function = async () => {
+				await old();
```

Comment:
Found this could be a little bit error-prone, as this hugely depends on what the hook type it is. Gotta find a way to fix this in the future.

Reply:
This is not needed once all hooks refactored 😄